js/index.js

```diff
@@ -0,0 +1,210 @@
+            close();
+        }
+        this.updateCount();
+    }
+
+    //double click 이벤트 - 수정하기
+    onEditingByDblClick = ({ target }) => {
+        if (!target) return;
+        if (target.tagName === "LABEL") {
+            this.previous = target.textContent;
+            target.parentElement.parentElement.classList.add('editing');
+        }
+        this.updateCount();
+    }
+
+    //todoItems 완료 및 삭제
+    onChangeToggle = ({ target }) => {
+        if (!target) return;
+        const cls = target.classList;
+        if (cls.contains("toggle")) {
+            let findTarget = target.parentElement.parentElement;
+            findTarget.classList.toggle('completed');
+            this.save(this.getTodoEntity(findTarget.dataset['idx']));
+            let currentTap = this.$todoFooter.querySelector(".selected");
+            currentTap.click();
+        } else if (cls.contains("destroy")) {
+            let findTarget = target.parentElement.parentElement;
+            this.delete(findTarget.dataset["idx"]);
+            findTarget.remove();
+            this.updateCount();
+        }
+    }
+
+    //하단 카운트 업데이트
+    updateCount = () => {
+        let count = this.$todoList.querySelectorAll("li:not(.blind)").length + "";
+        this.$todoFooter.querySelector(".todo-count>strong").textContent = count;
+    }
+
+    //하단 탭 변경 로직
+    switchTap = ({ target }) => {
+        const getCheckedItems = (on) => {
+            let resultList = [];
+            for (const item of this.$todoList.querySelectorAll(`li`)) {
+                if (item.querySelector(".toggle").checked === on) {
+                    resultList.push(item);
+                }
+            }
+            return resultList;
+        }
+        if (target && target.tagName === "A") {
+            let $todoItems = this.$todoList.getElementsByTagName("LI");
+            this.$todoFooter.querySelector(".selected").classList.remove('selected');
+            target.classList.add('selected');
+            let cls = target.classList;
+            if (cls.contains("all")) {
+                blinding($todoItems, false);
+            } else if (cls.contains("active")) {
+                blinding($todoItems, true);
+                blinding(getCheckedItems(false), false);
+            } else if (cls.contains("completed")) {
+                blinding($todoItems, true);
+                blinding(getCheckedItems(true), false);
+            }
+        }
+        this.updateCount();
+    }
+
+    //todoItme 추가 로직
+    onAddInput = ({ target, key }, check = false) => {
+        if (target && key === "Enter" && target.value !== "") {
+            const idx = this.getNextIndex();
+            this.$todoList.append(this.getFrame(idx, target.value, check));
+            this.save({idx, value:target.value, check});
+            target.value = '';
+            this.updateCount();
+        }
+    }
+
+    //todoItem Frame Maker
+    getFrame = (idx, value, check) => {
```

Comment:
```suggestion
    getFrame = ({ idx, value, check }) => {
```
이렇게 객체의 할당문으로 받아올 경우 객체를 그냥 넘겨주기만 하면 되서 더 편하고 효과적입니다.

Reply:
안녕하세요. 준일님.
리뷰 내용들을 읽다가 궁금한 것이 생겨서 질문드려봅니다.

혹시 그냥 값을 넘기는 것과 리뷰남겨주신 것처럼 `destructuring`으로 객체를 넘기는 것에는 어떤 차이가 있나요?

Reply:
@JSKim-Cian
item 객체를 바로 넘길 수 있다는 장점이 있습니다!
```js
cosnt item = { idx, value, check };
const getFrame = ({ idx, value, check }) => { /* 무언가 처리 */ }
getFrame(item);
```
이렇게요!